doc/rbd/rbd-live-migration.rst

```diff
@@ -91,6 +108,181 @@ usage during the migration process::
+            },
+            "snapshots": [
+                {
+                    "type": "raw",
+                    "name": "<snapshot-name>",
+                    "stream": {
+                        <stream unique parameters for snapshot>
+                    }
+                },
+            ] (optional oldest to newest ordering of snapshots)
+        }
+
+Support for snapshots is optional and currently only supports thick-provisioned
+``raw`` snapshot exports.
+
+Additional formats such as QCOW2, RBD export-format v2, and RBD export-diff
+snapshots will be added in a future release.
+
+Streams
+~~~~~~~
+
+The ``file`` stream can be used to import from a locally accessible POSIX file
+source. Its ``source-spec`` JSON is encoded as follows::
+
+        {
+            <format unique parameters>
+            "stream": {
+                "type": "file",
+                "file_path": "<file-path>"
+            }
+        }
+
+For example, to import a raw-format image from a file located at
+"/mnt/image.raw", its ``source-spec`` JSON is encoded as follows::
+
+        {
+            "type": "raw",
+            "stream": {
+                "type": "file",
+                "file_path": "/mnt/image.raw"
+            }
+        }
+
+The ``http`` stream can be used to import from a remote HTTP(s) web server.
+Its ``source-spec`` JSON is encoded as follows::
+
+        {
+            <format unique parameters>
+            "stream": {
+                "type": "http",
+                "url": "<url-path>"
+            }
+        }
+
+For example, to import a raw-format image from a file located at
+"http://download.ceph.com/image.raw", its ``source-spec`` JSON is encoded
+as follows::
+
+        {
+            "type": "raw",
+            "stream": {
+                "type": "http",
+                "url": "http://download.ceph.com/image.raw"
+            }
+        }
+
+The ``s3`` stream can be used to import from a remote S3 bucket. Its
+``source-spec`` JSON is encoded as follows::
+
+        {
+            <format unique parameters>
+            "stream": {
+                "type": "s3",
+                "url": "<url-path>",
+                "access_key": "<access-key>",
+                "secret_key": "<secret-key>"
+            }
+        }
+
+For example, to import a raw-format image from a file located at
+`http://s3.ceph.com/bucket/image.raw`, its ``source-spec`` JSON is encoded
+as follows::
+
+        {
+            "type": "raw",
+            "stream": {
+                "type": "s3",
+                "url": "http://s3.ceph.com/bucket/image.raw",
+                "access_key": "NX5QOQKC6BH2IDN8HC7A",
+                "secret_key": "LnEsqNNqZIpkzauboDcLXLcYaWwLQ3Kop0zAnKIn"
+            }
+        }
+
+.. note::
+  The ``access_key`` and ``secret_key`` parameters support storing the keys in
+  the MON config-key store by prefixing the key values with ``config://``
+  followed by the path in the MON config-key store to the value.
```

Comment:
nit: It would be nice to have a description (or link to) how to store a key in the MON config-key store.